backend/src/User/user.models.ts

```diff
@@ -36,13 +38,23 @@ class User {
             .first();
     }
 
+    public static async findByResetPasswordToken(token: string): Promise<User> {
+        return await db('user')
+            .select('userId', 'name', 'role', 'email', 'password')
+            .where('resetPasswordToken', token)
+            .andWhere('resetPasswordExpires', '>=', Date.now())
+            .first();
+    }
+
     public static async updateById(userID: number, user: User): Promise<number> {
         return await db('user')
             .update({
                 name: user.name,
                 role: user.role,
                 email: user.email,
-                password: user.password
+                password: user.password,
+                resetPasswordToken: user.resetPasswordToken,
+                resetPasswordExpires: user.resetPasswordExpires
```

Comment:
Not sure why we need token and expires. Shouldn't the expiration be with the token  already?

Reply:
No, the token is only a random string with no information

This way makes it easy to query in the db for a token and an expiration date. So there is no computation for creating a real token and decrypting that token